### PR TITLE
Improve peers data acquisition, logging, tuned peer sweeper

### PR DESCRIPTION
**Pull request by 4miners**
Closes https://github.com/LiskHQ/lisk/issues/435

### CODE REVIEW

File: modules/peers.js [hunk 1]
```diff
@@ -468,15 +484,45 @@ Peers.prototype.onPeersReady = function () {
...
+					if (err) {
+						library.logger.error('Get peers failed', err);
+						return setImmediate(seriesCb, err);
+					}
+
+					library.logger.trace('onPeersReady->updatePeers', {count: (peers ? peers.length : null)});
+					var updated = 0;
+					async.each(peers, function (peer, eachCb) {
+						library.logger.debug('Updating peer', {peer: peer});
+						// If peer is not banned and not been updated during last 3 sec - ping
```
Comment (4miners) [inline modules/peers.js]: That comment is bad, need to be removed.

File: modules/peers.js [hunk 2]
```diff
@@ -468,15 +484,45 @@ Peers.prototype.onPeersReady = function () {
...
 				__private.banManager(function (err) {
 					if (err) {
 						library.logger.error('Ban manager timer', err);
 					}
 					return setImmediate(seriesCb);
 				});
+			},
+			updatePeers: function (seriesCb) {
+				library.logger.trace('onPeersReady->updatePeers');
+				self.list ({limit: 500}, function (err, peers, consensus) {
```
Comment (karmacoma) [inline modules/peers.js]: `self.list({limit: 500}, function (err, peers, consensus) {`

File: modules/peers.js [hunk 3]
```diff
@@ -468,15 +484,45 @@ Peers.prototype.onPeersReady = function () {
...
+					var updated = 0;
+					async.each(peers, function (peer, eachCb) {
+						library.logger.debug('Updating peer', {peer: peer});
+						// If peer is not banned and not been updated during last 3 sec - ping
+						if (peer && peer.state > 0) {
+							self.pingPeer (peer, function (err, res) {
+								if (!err) {
+									++updated;
+								}
+								setImmediate(eachCb);
```
Comment (karmacoma) [inline modules/peers.js]: `return setImmediate(eachCb);`

File: modules/peers.js [hunk 4]
```diff
@@ -468,15 +484,45 @@ Peers.prototype.onPeersReady = function () {
...
+							});
+						} else {
+							setImmediate(eachCb);
```
Comment (karmacoma) [inline modules/peers.js]: `return setImmediate(eachCb);`

File: modules/peers.js [hunk 5]
```diff
@@ -468,15 +484,45 @@ Peers.prototype.onPeersReady = function () {
...
+	 					}
+					}, function () {
+						library.logger.trace('onPeersReady->updatePeers', {updated: updated, total: peers.length});
+						setImmediate(seriesCb);
```
Comment (karmacoma) [inline modules/peers.js]: `return setImmediate(seriesCb);`

File: modules/peers.js [hunk 6]
```diff
@@ -468,15 +484,45 @@ Peers.prototype.onPeersReady = function () {
...
+						return setImmediate(seriesCb, err);
+					}
+
+					library.logger.trace('onPeersReady->updatePeers', {count: (peers ? peers.length : null)});
+					var updated = 0;
+					async.each(peers, function (peer, eachCb) {
+						library.logger.debug('Updating peer', {peer: peer});
+						// If peer is not banned and not been updated during last 3 sec - ping
+						if (peer && peer.state > 0) {
+							self.pingPeer (peer, function (err, res) {
```
Comment (karmacoma) [inline modules/peers.js]: `self.pingPeer(peer, function (err, res) {`

File: modules/loader.js [hunk 7]
```diff
@@ -629,8 +629,10 @@ Loader.prototype.getNetwork = function (cb) {
 	return library.config.syncPeers.list.length ? setNetwork(library.config.syncPeers.list) : findDecentralizedPeers();
 
 	function findDecentralizedPeers () {
+		library.logger.trace('Finding decentralized peers');
 		async.waterfall([
 			function (waterCb) {
+				library.logger.trace('Get peers from random peer');
```
Comment (karmacoma) [inline modules/loader.js]: `Getting peers from random peer`

File: modules/loader.js [hunk 8]
```diff
@@ -643,6 +645,7 @@ Loader.prototype.getNetwork = function (cb) {
 				});
 			},
 			function (res, waterCb) {
+				library.logger.trace('Validate peers list');
```
Comment (karmacoma) [inline modules/loader.js]: `Validating peers list`

File: modules/peers.js [hunk 9]
```diff
@@ -408,6 +408,20 @@ Peers.prototype.update = function (peer) {
 	return __private.sweeper.push('upsert', self.accept(peer).object());
 };
 
+Peers.prototype.pingPeer = function (peer, cb) {
+	library.logger.trace('Ping peer: ' + peer.ip + ':' + peer.port);
```
Comment (karmacoma) [inline modules/peers.js]: `Pinging peer:`

File: modules/peers.js [hunk 10]
```diff
@@ -468,15 +484,45 @@ Peers.prototype.onPeersReady = function () {
...
 						library.logger.error('Ban manager timer', err);
 					}
 					return setImmediate(seriesCb);
 				});
+			},
+			updatePeers: function (seriesCb) {
+				library.logger.trace('onPeersReady->updatePeers');
+				self.list ({limit: 500}, function (err, peers, consensus) {
+					if (err) {
+						library.logger.error('Get peers failed', err);
```
Comment (karmacoma) [inline modules/peers.js]: `Peers listing failed`

**Comment by 4miners** [comment]
@karmacoma Fixed, thanks.

File: logic/peerSweeper.js [hunk 11]
```diff
@@ -5,18 +5,18 @@ var pgp = require('pg-promise');
...
+	this.limit = 500;
 	this.scope = scope;
 
 	var self = this;
 
 	setImmediate(function nextSweep () {
 		if (self.peers.length) {
 			self.sweep(self.peers.splice(0, self.limit), function () {
-				return setTimeout(nextSweep, 1000);
+				return setTimeout(nextSweep, 500);
```
Comment (MaciejBaj) [inline logic/peerSweeper.js]: so we can use this.limit variable here?

Comment (4miners) [inline logic/peerSweeper.js]: No, `this.limit` is for number of peers to sweep at once, and here we have time interval in ms.

Comment (karmacoma) [inline logic/peerSweeper.js]: They are now same value, but mean different things. `this.limit` is number of peers. However this is a time interval for next sweep.

**Pull request merged**